boards: arm: add support for WeAct STM32WB55 Core Board

heronet opened the pull request:
Add initial support for the WeAct Studio STM32WB55 Core Board based on the STM32WB55CGU6 dual-core MCU (Cortex-M4 + Cortex-M0+).

The board configuration supports:
 - 64 MHz system clock from 32 MHz HSE crystal
 - Console over USART1 (PA9/PA10)
 - Default LED on PB2
 - DFU and STM32CubeProgrammer flashing
 - External SWD debugging

Bluetooth functionality requires flashing a compatible STM32WB HCI Layer wireless coprocessor binary before use. The extended HCI binary (stm32wb5x_BLE_HCILayer_extended_fw.bin) has been verified.

Tested with:
 - samples/basic/blinky
 - samples/hello_world
 - samples/bluetooth/peripheral
 - samples/bluetooth/beacon

erwango reviewed (changes requested):
Please rework the doc part to make it clear when external probe and specifically ST-Link is required and when it is not.

heronet commented:
Apparently I created the branch off an older main so I did a rebase. Hope that is not going to be a problem.

erwango commented:
> Apparently I created the branch off an older main so I did a rebase. Hope that is not going to be a problem.

Sure you can. Please just take car of not force push mixing rebase with new changes. You've indeed done it the right way, thanks

heronet committed:
boards: arm: add support for WeAct STM32WB55 Core Board

Add initial support for the WeAct Studio STM32WB55 Core Board based on the
STM32WB55CGU6 dual-core MCU (Cortex-M4 + Cortex-M0+).

The board configuration supports:
 - 64 MHz system clock from 32 MHz HSE crystal
 - Console over USART1 (PA9/PA10)
 - Default LED on PB2
 - DFU and STM32CubeProgrammer flashing
 - External SWD debugging

Bluetooth functionality requires flashing a compatible STM32WB HCI Layer
wireless coprocessor binary before use. The extended HCI binary
(stm32wb5x_BLE_HCILayer_extended_fw.bin) has been verified.

Tested with:
 - samples/basic/blinky
 - samples/hello_world
 - samples/bluetooth/peripheral
 - samples/bluetooth/beacon

Signed-off-by: Siratul Islam <sirat4757@gmail.com>
